src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs

```diff
@@ -314,173 +499,1016 @@ void ValidateDisposedException(Exception e, bool successAllowed = false)
             }
         }
 
-        protected async Task ValidatePrecanceledOperations_ThrowsCancellationException(Stream stream)
+        protected async Task ValidatePrecanceledOperations_ThrowsCancellationException(Stream stream)
+        {
+            var cts = new CancellationTokenSource();
+            cts.Cancel();
+
+            OperationCanceledException oce;
+
+            if (stream.CanRead)
+            {
+                oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => stream.ReadAsync(new byte[1], 0, 1, cts.Token));
+                Assert.Equal(cts.Token, oce.CancellationToken);
+
+                oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => { await stream.ReadAsync(new Memory<byte>(new byte[1]), cts.Token); });
+                Assert.Equal(cts.Token, oce.CancellationToken);
+            }
+
+            if (stream.CanWrite)
+            {
+                oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => stream.WriteAsync(new byte[1], 0, 1, cts.Token));
+                Assert.Equal(cts.Token, oce.CancellationToken);
+
+                oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => { await stream.WriteAsync(new ReadOnlyMemory<byte>(new byte[1]), cts.Token); });
+                Assert.Equal(cts.Token, oce.CancellationToken);
```

Comment:
Just a thought: Maybe we should have some sort of Assert for cancellation specifically, that also checks the cancellationToken? Seems like this is a pretty common pattern here.

Reply:
Added a helper.